packages/swing-store/src/swingStore.js

```diff
@@ -383,21 +555,20 @@ function makeSwingStore(dirPath, forceReset, options) {
  * undefined, a memory-only ephemeral store will be created that will evaporate
  * on program exit.
  *
- * @param {string|null} dirPath  Path to a directory in which database files may
+ * @param {string|null} dirPath Path to a directory in which database files may
  *   be kept.  This directory need not actually exist yet (if it doesn't it will
  *   be created) but it is reserved (by the caller) for the exclusive use of
- *   this swing store instance.  If null, an ephemeral store will be created.
+ *   this swing store instance.  If null, an ephemeral (memory only) store will
+ *   be created.
  * @param {object?} options  Optional configuration options
  *
  * @returns {SwingStore}
  */
-export function initSwingStore(dirPath, options = {}) {
-  if (!dirPath) {
-    return initEphemeralSwingStore();
-  } else {
+export function initSwingStore(dirPath = null, options = {}) {
```

Comment:
I don't like that design.  Most use cases really don't want to have to care if the DB already exists or not, they just want to get a store back.  What they _do_ care about is "start from scratch, regardless of the previous state of the world" vs. "continue where you last left off from, even that if that was t0".

Reply:
Let's chat further (outside the scope of this PR). I think most applications *should* care about existing/not-existing: if they don't care, they're doing it wrong, and the result is an easier path to accidents (accidentally clobbering your precious state directory because you re-ran the `init` subcommand, or getting inconsistent behavior out of your `run` subcommand. Swingset host applications have a "initialize a blank directory" step, run once in e.g. the entire history of the chain, and then one or more "launch the application from the already-initialized directory" steps, which they do every time they reboot. That maps directly onto the API I advocated. And I think "sometimes create, sometimes re-use existing" is an anti-pattern that is more useful for test cases and docs than real applications. (Mind you, I've changed my opinion on this multiple times over the course of my career). I'll file an issue.